packages/dev/core/src/Particles/Node/Blocks/Update/updateNoiseBlock.ts

```diff
@@ -0,0 +1,137 @@
+import type { NodeParticleConnectionPoint } from "core/Particles/Node/nodeParticleBlockConnectionPoint";
+import type { NodeParticleBuildState } from "core/Particles/Node/nodeParticleBuildState";
+import type { INodeParticleTextureData, ParticleTextureSourceBlock } from "core/Particles/Node/Blocks/particleSourceTextureBlock";
+
+import { TmpVectors, Vector3 } from "core/Maths/math.vector";
+import { RegisterClass } from "core/Misc/typeStore";
+import { NodeParticleBlock } from "core/Particles/Node/nodeParticleBlock";
+import { NodeParticleBlockConnectionPointTypes } from "core/Particles/Node/Enums/nodeParticleBlockConnectionPointTypes";
+import { _ConnectAtTheEnd } from "core/Particles/Queue/executionQueue";
+
+/**
+ * Block used to update particle position based on a noise texture
+ */
+export class UpdateNoiseBlock extends NodeParticleBlock {
+    /**
+     * Create a new UpdateNoiseBlock
+     * @param name defines the block name
+     */
+    public constructor(name: string) {
+        super(name);
+
+        this.registerInput("particle", NodeParticleBlockConnectionPointTypes.Particle);
+        this.registerInput("noiseTexture", NodeParticleBlockConnectionPointTypes.Texture);
+        this.registerInput("strength", NodeParticleBlockConnectionPointTypes.Vector3, true, new Vector3(100, 100, 100));
+        this.registerOutput("output", NodeParticleBlockConnectionPointTypes.Particle);
+    }
+
+    /**
+     * Gets the particle component
+     */
+    public get particle(): NodeParticleConnectionPoint {
+        return this._inputs[0];
+    }
+
+    /**
+     * Gets the noiseTexture input component
+     */
+    public get noiseTexture(): NodeParticleConnectionPoint {
+        return this._inputs[1];
+    }
+
+    /**
+     * Gets the strength input component
+     */
+    public get strength(): NodeParticleConnectionPoint {
+        return this._inputs[2];
+    }
+
+    /**
+     * Gets the output component
+     */
+    public get output(): NodeParticleConnectionPoint {
+        return this._outputs[0];
+    }
+
+    /**
+     * Gets the current class name
+     * @returns the class name
+     */
+    public override getClassName() {
+        return "UpdateNoiseBlock";
+    }
+
+    /**
+     * Builds the block
+     * @param state defines the current build state
+     */
+    public override _build(state: NodeParticleBuildState) {
+        const system = this.particle.getConnectedValue(state) as ThinParticleSystem;
+
+        const strength = this.strength.getConnectedValue(state) as Vector3;
+        if (!strength) {
+            return;
+        }
+
+        const noiseTexture = this.noiseTexture.connectedPoint?.ownerBlock as ParticleTextureSourceBlock;
+        if (!noiseTexture) {
+            return;
+        }
+
+        let noiseData: INodeParticleTextureData;
+
+        // eslint-disable-next-line github/no-then
+        void noiseTexture.extractTextureContentAsync().then((textureContent: Nullable<INodeParticleTextureData>) => {

```

Comment:
This needs to be call on every frame right? Unless I'm wrong (which is totally possible). This will be called just once and never change?

Reply:
Yeah this needs to go inside the process noise function

Reply:
Addressed.